### DISCUSSION TITLE
Using secret provider class in Kustomize

### DISCUSSION THREAD

**Discussion by monicabiswal147** [kubernetes-sigs/kustomize#5418] [Q&A]
Hi team,

When we do the kubernetes integration with vault using CSI provider, we usually define a secret provider class(SPC) which contains all the information about the secrets present in vault which is then volume mounted on to a pod. Does Kustomize provides the support for using SPC in the secrets generator? We do not want to define the secrets again in kustomization.yaml or Is there any other way to achieve this using Kustomize?

[a comment by an automated account is not shown]

**Comment by annasong20** [comment]
@monicabiswal147 I don't have prior experience with vault, CSI provider, or SPCs, but from skimming https://kubernetes.io/blog/2019/01/15/container-storage-interface-ga/, it sounds like you're asking if Kustomize's `secretGenerator` interfaces with external providers to fetch secrets. If this is the case, no, Kustomize's [secretGenerator](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/secretgenerator/) is more rudimentary than that. It generates secrets from literals or files, at most base-64 encoding or adding labels to the generated secret, etc.

We recommend asking questions in the Kustomize Slack channel
https://kubernetes.slack.com/archives/C9A5ALABG or starting a GitHub Discussion. You're likely to get a faster response that way!
